src/providers/incoming-data/incoming-data.ts

```diff
@@ -802,11 +821,29 @@ export class IncomingDataProvider {
     this.events.publish('IncomingDataRedir', nextView);
   }
 
-  public getPayProDetails(data: string): Promise<any> {
-    let coin: string = data.indexOf('bitcoincash') === 0 ? Coin.BCH : Coin.BTC;
-    data = decodeURIComponent(data.replace(/bitcoin(cash)?:\?r=/, ''));
-
-    let disableLoader = true;
-    return this.payproProvider.getPayProDetails(data, coin, disableLoader);
+  public getPayProDetails(paymentUrl: string): Promise<any> {
+    let coin;
+    if (paymentUrl.indexOf('bitcoincash') === 0) {
+      coin = Coin.BCH;
+      paymentUrl = decodeURIComponent(
+        paymentUrl.replace(/bitcoin(cash)?:\?r=/, '')
+      );
+    }
+    if (paymentUrl.indexOf('bitcoin') === 0) {
+      coin = Coin.BTC;
+      paymentUrl = decodeURIComponent(
+        paymentUrl.replace(/bitcoin(cash)?:\?r=/, '')
+      );
+    }
+    if (paymentUrl.indexOf('ethereum') === 0) {
+      coin = Coin.ETH;
+      paymentUrl = decodeURIComponent(paymentUrl.replace(/ethereum?:\?r=/, ''));
+    }
```

Comment:
Use `getCoinFromUri` and `getPayProUrl` functions here to not have any merge conflicts into eth branch.

https://github.com/bitpay/copay/pull/10179/files#diff-879a6f0c0c6622976ede2f66f41ab9f6R730-R755